compiler-rt/test/tsan/free_race_aligned_alloc.c

```diff
@@ -0,0 +1,61 @@
+// RUN: %clang_tsan -O1 %s -o %t -undefined dynamic_lookup
+// RUN: %deflake %run %t | FileCheck %s
+
+#include "test.h"
+
+#include <stdlib.h>
+
+#if defined(__cplusplus) && (__cplusplus >= 201703L)
+#  define HAVE_ALIGNED_ALLOC 1
+#endif
+
+#if defined(__ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__) &&                  \
```

Comment:
instead of using `-undefined`, can you make #if #elif ... to avoid `#define HAVE_ALIGNED_ALLOC` conflicts?

Reply:
I refactored such that we only define `HAVE_ALIGNED_ALLOC` one time with no conflicts. Let me know if that isn't what you intended!